integration-tests/src/test/java/com/datastax/oss/driver/core/cql/PreparedStatementIT.java

```diff
@@ -392,10 +391,15 @@ public void should_create_separate_instances_for_differently_formatted_queries()
       assertThat(getPreparedCacheSize(session)).isEqualTo(0);
 
       // When
+      CompletionStage<PreparedStatement> preparedStatement1Future =
+          session.prepareAsync("SELECT * FROM prepared_statement_test WHERE a = ?");
+      CompletionStage<PreparedStatement> preparedStatement2Future =
+          session.prepareAsync("select * from prepared_statement_test where a = ?");
+
       PreparedStatement preparedStatement1 =
-          session.prepare("SELECT * FROM prepared_statement_test WHERE a = ?");
+          CompletableFutures.getUninterruptibly(preparedStatement1Future);
       PreparedStatement preparedStatement2 =
-          session.prepare("select * from prepared_statement_test where a = ?");
+          CompletableFutures.getUninterruptibly(preparedStatement2Future);
```

Comment:
So I'm a little confused about this.

The original test(s) include an initial check to make sure the cache is empty at first, followed by checks to make sure reprepare either returns the same instance (if it's the same text) or something different (if the text differs slightly).  We then validate the cache size to see if it matches expectations.

I'm a bit confused as to what the point of the cache size checks are.  The whole point of this test is to confirm whether we get back the same instance for the same text (or a different instance for slightly different text).  Why is the cache size relevant there?  Feels like that's an implementation detail.  If the driver returns the same instance for the same query this test should pass; whether it uses a cache or not is irrelevant.

I understand you didn't write this test and that you're only modifying what was there to avoid the discussed GC issue.  But as I mentioned when we first talked about this I don't _love_ the preservation of the CompletionStages above to short-circuit the weak ref cache vals.  That led me to wonder whether we really need them here... which in turn led me to wonder whether the cache size checks really make sense here.

Reply:
If we're going to keep the CompletionStages around we should at least add a comment saying this represents an explicit design choice and explain why we're doing it.  Said ticket should probably reference the JIRA ticket as well.